File: hackerrank/problem solving/algorithms/2 - implementation/24 - viral advertising/viral_advertising_procedural.cpp
// Source: https://www.hackerrank.com/challenges/strange-advertising/problem?isFullScreen=true

#include <iostream>

using namespace std;

int viral_advertising(const int days);


int main() {
	int days;
	cin >> days;

	const int cumulative_likes {viral_advertising(days)};
	cout << cumulative_likes << "\n";

	return 0;
}

	int viral_advertising(const int days) {
		int shared {5}, liked {0}, cumulative {0};

		for (int _ {0}; _ < days; _++) {
			liked = {shared / 2};
			cumulative += liked;
			shared = {liked * 3};
		}

		return cumulative;
	}

File: hackerrank/problem solving/algorithms/2 - implementation/24 - viral advertising/viral_advertising_procedural.py
# Source: https://www.hackerrank.com/challenges/strange-advertising/problem?isFullScreen=true

def main():

    days = int(input())
    cumulativeLikes = viralAdvertising(days)
    print(cumulativeLikes)


def viralAdvertising(days):

    shared, liked, cumulative = 5, 0, 0

    for _ in range(days):
        liked = int(shared / 2)
        cumulative += liked
        shared = liked * 3

    return cumulative


if __name__ == "__main__":
    main()
